internal/metrics/metrics_test.go

```diff
@@ -0,0 +1,69 @@
+package metrics
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	dto "github.com/prometheus/client_model/go"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	numberOfRegisteredValue       = 3
+	numberOfFailedToRegisterValue =1
+	numberOfCreatedOBCsValue      =2
+)
+
+func TestMetrics(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Metrics Suite")
+}
+
+func findMetric(src []*dto.MetricFamily, query string) *dto.MetricFamily {
+	for _, s := range src {
+		if s.Name != nil && *s.Name == query {
+			return s
+		}
+	}
+	return nil
+}
+
+var _ = Describe("Metrics", func() {
+	m := New()
+	for  i := 0; i < numberOfRegisteredValue; i++ {
+		m.IncRegisteredEdgeDevice()
+	}
+	for  i := 0; i < numberOfFailedToRegisterValue; i++ {
+		m.IncFailedRegistrationEdgeDevice()
+	}
+	for  i := 0; i < numberOfCreatedOBCsValue; i++ {
+		m.IncCreatedOBCs()
+	}
+
+	It("correctly passes calls to the collectors", func() {
```

Comment:
Please use 
```
// given
// when
// then 
```
template for tests to highlight their different aspects (input, execution, verification)